oaknational/Oak-Web-Application
# LESQ-1045: implement filter drawer on unit pages

## Description

Music year: 2022

Implements filter drawer component on specialist and non-specialist unit listing pages

- Creates `MobileUnitFilters` component and test file
- Renders filter drawer on Specialist and Non-Specialist unit listing pages 


## Acceptance Criteria:

- [x]  Make the search filter drawer and the unit listing drawer the same component
Filter drawer component created here: https://github.com/oaknational/oak-components/pull/291 not implemented on Search as of yet 
- [x]  Close, clear and (sticky) show results buttons
- [x]  Dropdown shouldn’t close until a user interacts with the close or show results buttons
    - Close cancels the action and doesn’t apply the selected filters, closes the dropdown
    - Show results applies the selected filters, closes the dropdown
- [x]  Number of results should be on the apply button

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2835--oak-web-application.netlify.thenational.academy
https://deploy-preview-2835--oak-web-application.netlify.thenational.academy/teachers/programmes/combined-science-secondary-ks4-foundation-edexcel/units
2. Navigate to any unit page with filters
3. You should see on mobile and tablet view filter button is rendered, when selecting this, the FilterDrawer should open
4. When clicking the filters, selected filter is highlighted, the draw doesn't close, the submit button is update with number of units
5. Clear button clears all the filters, X button closes the filter drawer without applying changes, submit button updates query and correct units displayed.

## Screenshots

How it should now look:
(Re-uploading svg on filter button)

https://github.com/user-attachments/assets/4149b8c6-34db-4dd3-8043-f884104d8f16



## Checklist
